### PR TITLE
Upgrade core packages

### PR DESCRIPTION
**Pull request by seallard**
## Description
This PR upgrades the versions of these packages:
- Flask 1.1.4 -> 3.0.0 [change log](https://flask.palletsprojects.com/en/3.0.x/changes/)
- Flask-Dance 5.1.0 -> 7.0.0 [change log](https://github.com/singingwolfboy/flask-dance/blob/main/CHANGELOG.rst)
- Click 6.7 -> 8.1.7 [change log](https://click.palletsprojects.com/en/8.1.x/changes/)

Each change log was read and changes made accordingly (for any breaking changes).

It is necessary to upgrade all of them at once since they lock each other. Upgrading these packages has the upside of also letting us upgrade Python, they all support Python 3.12.

## Upgrading flask 1.1.4 -> 3.0.0 
- Replace `from flask.json import JSONEncoder` with `from json import JSONEncoder`, [removed in 2.3.0](https://flask.palletsprojects.com/en/2.3.x/changes/#version-2-3-0).
- Replace `json_encoder` with `json_provider_class` on app, [removed in 2.3.0](https://flask.palletsprojects.com/en/3.0.x/changes/#version-2-3-0).
- Replace `file_name` argument with `path` in `send_from_directory`, [renamed in 2.2.0](https://flask.palletsprojects.com/en/3.0.x/changes/#version-2-2-0).
- Remove env variable for deprecated attribute `templates_auto_reload`, [removed in 2.3.0](https://flask.palletsprojects.com/en/3.0.x/changes/#version-2-3-0).

Notes:
- `SESSION_COOKIE_DOMAIN` is no longer set by default, see [this](https://github.com/pallets/flask/issues/5051) and [change log for 2.3.0](https://flask.palletsprojects.com/en/3.0.x/changes/#version-2-3-0). It does not seem to matter though.

## Upgrading flask-dance 5.1.0 -> 7.0.0
No breaking changes beyond requiring a more recent version of Flask, see [change log](https://github.com/singingwolfboy/flask-dance/blob/main/CHANGELOG.rst).

## Upgrading Click 6.7 -> 8.1.7 
No relevant breaking changes, see [change log](https://click.palletsprojects.com/en/8.1.x/changes/).

### Fixed
- Upgrade flask, flask-dance and click to the latest versions.

### This [version](https://semver.org/) is a
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan
- [x] Upgrade to Python 3.8 from 3.7 in cg and housekeeper (Flask 2 [dropped support for Python 3.7](https://flask.palletsprojects.com/en/2.3.x/changes/#version-2-3-0).)
- [x] Upgrade packages in housekeeper
- [x] Test this branch in stage on hasta and vm
- [ ] Merge this PR

### CODE REVIEW

**Comment by seallard** [comment]
Error in status db view:
```
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]: AttributeError: type object 'CustomJSONEncoder' has no attribute 'dumps'
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return current_app.json.dumps(obj, **kwargs)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/json/__init__.py", line 41, in dumps
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return dumps(self.tag(value), separators=(",", ":"))
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/json/tag.py", line 310, in dumps
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return isinstance(serializer.dumps({}), str)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/itsdangerous/serializer.py", line 22, in is_text_serializer
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     self.is_text_serializer: bool = is_text_serializer(serializer)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/itsdangerous/serializer.py", line 116, in __init__
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return URLSafeTimedSerializer(
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/sessions.py", line 301, in get_signing_serializer
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     s = self.get_signing_serializer(app)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/sessions.py", line 309, in open_session
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     self.session = session_interface.open_session(self.app, self.request)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/ctx.py", line 377, in push
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     ctx.push()
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1454, in wsgi_app
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return cors_after_request(app.make_response(f(*args, **kwargs)))
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask_cors/extension.py", line 176, in wrapped_function
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     response = self.handle_exception(e)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1458, in wsgi_app
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     return self.wsgi_app(environ, start_response)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/flask/app.py", line 1478, in __call__
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     respiter = self.wsgi(environ, resp.start_response)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/gunicorn/workers/gthread.py", line 334, in handle_request
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:     keepalive = self.handle_request(req, conn)
Oct 05 16:44:24 cg-vm1.scilifelab.se podman[91113]:   File "/usr/local/lib/python3.8/site-packages/gunicorn/workers/gthread.py", line 282, in handle
```

File: requirements.txt
Comment (diitaz93) [inline requirements.txt:7]: 🙌

**Review by diitaz93 — APPROVED** [review]
Looks good! Maybe wait until Monday to merge it just in case

**Comment by diitaz93** [comment]
Would we need to do anything in our local environments when this is merged?

**Comment by seallard** [comment]
> Would we need to do anything in our local environments when this is merged?

Yes, will have to do a manual pip upgrade of these packages in the cg conda environment on Hasta since it picks up the cached libraries by default

**Review by henrikstranneheim — APPROVED** [review]
Awesome 💯

**Comment by seallard** [comment]
I'll deploy it now and monitor during the rest of the day. In case of any issues, I'll revert it

**Pull request merged**
